src/test/fuzz/txorphan.cpp

```diff
@@ -55,10 +55,27 @@ FUZZ_TARGET(txorphan, .init = initialize_orphanage)
             const auto num_out = fuzzed_data_provider.ConsumeIntegralInRange<uint32_t>(1, 256);
             // pick outpoints from outpoints as input. We allow input duplicates on purpose, given we are not
             // running any transaction validation logic before adding transactions to the orphanage
-            for (uint32_t i = 0; i < num_in; i++) {
-                auto& prevout = PickValue(fuzzed_data_provider, outpoints);
-                // try making transactions unique by setting a random nSequence, but allow duplicate transactions if they happen
-                tx_mut.vin.emplace_back(prevout, CScript{}, fuzzed_data_provider.ConsumeIntegralInRange<uint32_t>(0, CTxIn::SEQUENCE_FINAL));
+            {
```

Comment:
Since a random access vector seems to resolve the problem, is there any reason we cannot change the `outpoints` definition to a vector instead?

Reply:
I means simply:
```patch
===================================================================
diff --git a/src/test/fuzz/txorphan.cpp b/src/test/fuzz/txorphan.cpp
--- a/src/test/fuzz/txorphan.cpp	(revision 37992244e636e52e0c2baff1bc5f36e60d9c956f)
+++ b/src/test/fuzz/txorphan.cpp	(date 1721306232235)
@@ -37,11 +37,11 @@
     SetMockTime(ConsumeTime(fuzzed_data_provider));
 
     TxOrphanage orphanage;
-    std::set<COutPoint> outpoints;
+    std::vector<COutPoint> outpoints;
 
     // initial outpoints used to construct transactions later
     for (uint8_t i = 0; i < 4; i++) {
-        outpoints.emplace(Txid::FromUint256(uint256{i}), 0);
+        outpoints.emplace_back(Txid::FromUint256(uint256{i}), 0);
     }
 
     CTransactionRef ptx_potential_parent = nullptr;
@@ -67,7 +67,7 @@
             auto new_tx = MakeTransactionRef(tx_mut);
             // add newly constructed outpoints to the coin pool
             for (uint32_t i = 0; i < num_out; i++) {
-                outpoints.emplace(new_tx->GetHash(), i);
+                outpoints.emplace_back(new_tx->GetHash(), i);
             }
             return new_tx;
         }();
@@ -147,12 +147,10 @@
                     orphanage.LimitOrphans(limit, limit_orphans_rng);
                     Assert(orphanage.Size() <= limit);
                 });
-
         }
         // Set tx as potential parent to be used for future GetChildren() calls.
         if (!ptx_potential_parent || fuzzed_data_provider.ConsumeBool()) {
             ptx_potential_parent = tx;
         }
-
     }
 }
```